### DISCUSSION TITLE
OpenTopography maps

### DISCUSSION THREAD

**Discussion by Theis76** [gpx-animator/gpx-animator#485] [Q&A]
I'm trying to use OpenTopo maps but at they kept getting slower and slower I decided to try to use them dirctly with an APIkey. Opentopography provides an example string that looks like:
"https://portal.opentopography.org/API/globaldem?demtype=NASADEM&south=45.196&north=49&west=-122.66&east=-119.95&outputFormat=GTiff&API_Key=MYAPIKEY"

which does not look much like the GPX animator example: 
"https://map.provider.domain/path/{zoom}/{x}/{y}.png?apikey=ABC123"

How do I call portal.opentopography.org correctly?

**Comment by McPringle** [comment]
> How do I call portal.opentopography.org correctly?

You need to specify the TMS URL. TMS is a standard ([Tile Map Service](https://en.wikipedia.org/wiki/Tile_Map_Service)). Check the documentation of your provider or ask him. For OpenTopoMap for example, the TMS URL looks like this:

`https://{switch:a,b,c}.tile.opentopomap.org/{zoom}/{x}/{y}.png`

**Reply by nagelp** [reply]
I'm having the same issue, and I'd like to point out that you didn't actually answer the question on how to speed up tile download for OpenTopo maps by using an API key. I've requested one from opentopography.org as well, but when I specify it in the GUI that doesn't change anything.

I'm aware that this kind of bulk-downloading is discouraged by the tile server operators. Do you think it's a lot of effort to change GPX Animator to download the tiles on demand?

Also, the upper limit for the download cache lifetime seems to coincide with what OSM says should be the lower limit (https://operations.osmfoundation.org/policies/tiles/). At least I'd like to keep my cache longer than 9 days.

I've even briefly considered to run my own tile server (https://github.com/Overv/openstreetmap-tile-server/blob/master/README.md) but that seems too much effort.

**Reply by McPringle** [reply]
Sorry for my late reply, the last few months I took a sabbatical and rode my bicycle through the whole US. I hope you did not stop using GPX Animator in the meantime!

> I'm having the same issue, and I'd like to point out that you didn't actually answer the question

Well, @Theis76 asked about how to access OpenTopo maps. The example link he provided does not look like a TMS URL, so I told GPX Animator needs a TMS URL.

> on how to speed up tile download for OpenTopo maps by using an API key

The speed is sometimes reduced (depending on the server) when users are not logged in. Using an API key identifies the user so the server knows who it is and if he/she is given more bandwidth.

> I'm aware that this kind of bulk-downloading is discouraged by the tile server operators.

What GPX Animator does is not really bulk downloading. It downloads exactly only what is needed. That is exactly the same any browser does when viewing the same map.

> Do you think it's a lot of effort to change GPX Animator to download the tiles on demand?

That is exactly what GPX is doing. It downloads the map tiles only when needed. The map is the same for the whole video, it is rendered only once in the beginning to be put in the background of every frame of the video. The tiles for this one map are only downloaded when they are not already cached.

> Also, the upper limit for the download cache lifetime seems to coincide with what OSM says should be the lower limit

There is no upper limit of cache lifetime in GPX Animator. You can specify any value you like. I changed the default to 7 days, thanks for the hint. But GPX Animator does not depend on OSM. It can use all TMS servers available. They all have different recommendations of caching. That's why it is configurable in the preferences of GPX Animator.

> At least I'd like to keep my cache longer than 9 days.

Open the settings (File > Preferences) and specify any caching time you like.

> I've even briefly considered to run my own tile server but that seems too much effort.

Well, for most people it is more about space. The database alone is ~1.5 TB and for storing the tiles you need ~70 TB. You can configure a tile server to render the tiles on request, but that is - of course - much, much slower.
